From aea3842ac1e52534ebde945d979f638b15387550 Mon Sep 17 00:00:00 2001
From: Brian Bennett <brian.bennett@joyent.com>
Date: Mon, 19 Feb 2018 16:46:43 -0800
Subject: [PATCH] OS-6518 rsyslog needs audit facility support Reviewed by:
 Jason King <jason.king@joyent.com> Reviewed by: Dan McDonald
 <danmcd@joyent.com> Approved by: Dan McDonald <danmcd@joyent.com>

---
 rsyslog/Makefile                              |  2 ++
 rsyslog/patches/0001-illumos-facilities.patch | 15 +++++++++++++++
 2 files changed, 17 insertions(+)
 create mode 100644 rsyslog/patches/0001-illumos-facilities.patch

diff --git a/rsyslog/Makefile b/rsyslog/Makefile
index 3859733..ff4c478 100644
--- a/rsyslog/Makefile
+++ b/rsyslog/Makefile
@@ -31,6 +31,8 @@ AUTOCONF_OPTS += \
 	--enable-mail=yes \
 	--enable-omprog=yes
 
+PATCHES = patches/*
+
 include ../Makefile.targ
 
 all: all_autoconf
diff --git a/rsyslog/patches/0001-illumos-facilities.patch b/rsyslog/patches/0001-illumos-facilities.patch
new file mode 100644
index 0000000..b96dede
--- /dev/null
+++ b/rsyslog/patches/0001-illumos-facilities.patch
@@ -0,0 +1,15 @@
+--- a/runtime/srutils.c	2018-02-19 01:59:24.000000000 -0800
++++ b/runtime/srutils.c	2018-02-19 02:04:38.000000000 -0800
+@@ -94,6 +94,12 @@
+ #if defined(LOG_FTP)
+ 	{"ftp",          LOG_FTP},
+ #endif
++#if defined(__sun)
++	{"audit",        LOG_AUDIT},
++	{"altcron",      LOG_ALTCRON},
++	{"console",      LOG_CONSOLE},
++	{"ntp",          LOG_NTP},
++#endif
+ 	{"local0",       LOG_LOCAL0},
+ 	{"local1",       LOG_LOCAL1},
+ 	{"local2",       LOG_LOCAL2},
-- 
2.21.0

